paperslicer/media/extractor.py

```
@@ -0,0 +1,93 @@
+        out: List[Dict] = []
+        for pno in range(len(doc)):
+            page = doc[pno]
+            for j, info in enumerate(page.get_images(full=True), start=1):
+                xref = info[0]
+                pix = fitz.Pixmap(doc, xref)
+                if pix.n > 4:
+                    pix = fitz.Pixmap(fitz.csRGB, pix)
+                img_name = f"page{pno+1:03d}_img{j:02d}.png"
+                img_path = os.path.join(outdir, img_name)
+                pix.save(img_path)
+                sha1 = hashlib.sha1(open(img_path, "rb").read()).hexdigest()[:8]
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)